frontends/concrete-python/concrete/fhe/compilation/module_compiler.py

```diff
@@ -506,6 +507,23 @@ def compile(
 
         dbg = DebugManager(configuration)
 
+        # if using tfhers integers, parameter selection strategy has to be multi-parameters.
+        # we try to catch this early, but the compiler will also fail and provide some info about
+        # it. So we don't need to scan all the inputset here.
+        if inputsets is not None:
+            for input_set in inputsets.values():
+                if (
+                    check_inputset_has_element_of_type(input_set, TFHERSInteger, search_all=False)
+                    and self.configuration.parameter_selection_strategy
+                    != ParameterSelectionStrategy.MULTI
+                ):
+                    msg = (
+                        "Can't use tfhers integers with "
+                        f"{self.configuration.parameter_selection_strategy} parameters. "
+                        "Please use multi-parameters instead."
+                    )
+                    raise RuntimeError(msg)
```

Comment:
I feel like we can ignore it here and check for it during conversion. The problem with doing it here is that we'd need to duplicate this check for direct circuits as well in the future.

Reply:
(same with compiler.py)

Reply:
I moved it to conversion